plaso/parsers/sqlite.py

```diff
@@ -418,6 +426,20 @@ def ParseFileEntry(self, parser_mediator, file_entry):
         if not plugin.REQUIRED_TABLES.issubset(table_names):
           continue
 
+        columns_match = True
```

Comment:
Maybe move this into a `_CheckRequiredTablesAndColumns` method?

Reply:
I've moved this to a method. I had kept the REQUIRED_TABLES part untouched, as it was, but now think that might not be best. What do you think about combining REQUIRED_TABLES and REQUIRED_COLUMNS into just one check? I think it makes sense and would be less duplicative for plugin authors and slightly more efficient when running. I'll need to update the REQUIRED_x definitions in each plugin though, so I'd like to get sign-off first.

Reply:
> What do you think about combining REQUIRED_TABLES and REQUIRED_COLUMNS into just one check?

SG. Though I'm thinking would there be plugins that only would have REQUIRED_TABLES?

Reply:
Yes, there are some cases where checking for the presence of a table is needed to help identify an otherwise generically-named database, even if we don't use any data from that table. In those cases, the table name is given as required, and the required columns within it is an empty set (basically, we don't care about the columns).

Here is an example where specifying a table name is helpful, even though it isn't used in a query (`android_metadata`), because otherwise just checking for a table named `cookies` is too weak and error-prone:
```
  REQUIRED_STRUCTURE = {
      'android_metadata': frozenset([]),
      'cookies': frozenset([
          '_id', 'name', 'value', 'domain', 'expires', 'path', 'secure'])}
```

I've went and updated all the SQLite plugins to use REQUIRED_STRUCTURE. In most cases, I've added a bit stronger checks to the requirements, which should help avoid running the wrong plugin against a file, but still let the plugin run if minor things about a database change.